lib/OpenQA/Worker/Job.pm

```diff
@@ -1077,9 +1079,11 @@ sub _upload_asset {
                 local => $local_upload,
                 retries => $retries
             });
-    };
-    log_error($@, channels => \@channels_both, default => 1) if $@;
-
+    }
+    catch ($e) {
+        log_error($e, channels => \@channels_both, default => 1);
+        return 0;
```

Comment:
I think the `$ua->upload->unsubscribe($_)` below was always executed even if there was an error, right? Now it's returning before that